OJS.Workers.ExecutionStrategies/NodeJs/NodeJsProjectRunJavaScriptProjectAndTestsWithPlaywrigth.cs

```diff
@@ -0,0 +1,279 @@
+                    'mode': 'ro',
+                }},
+                path_to_project: {{
+                    'bind': '/usr/share/nginx/html',
+                    'mode': 'rw',
+                }},
+                path_to_node_modules: {{
+                    'bind': '/usr/share/nginx/html/node_modules',
+                    'mode': 'ro'
+                }}
+            }},
+        )
+
+    def start(self):
+        self.container.start()
+
+    def stop(self):
+        self.container.stop()
+        self.container.wait()
+        self.container.remove()
+
+
+executor = DockerExecutor()
+
+executor.start()
+
+commands = [mocha_path, tests_path, '-R', 'json']
+
+process = subprocess.run(
+    commands,
+    capture_output=True
+)
+
+print(process.stdout)
+
+executor.stop()
+
+";
+
+        protected string NginxFileContent => $@"
+worker_processes  1;
+
+events {{
+    worker_connections  1024;
+}}
+
+http {{
+    include mime.types;
+    types
+    {{
+        application/javascript mjs;
+    }}
+
+    default_type  application/octet-stream;
+    sendfile        on;
+    keepalive_timeout  65;
+
+    server {{
+      root /usr/share/nginx/html;
+
+      listen 80;
+      server_name localhost;
+
+      location / {{
+        try_files $uri $uri/ /index.html;
+        autoindex on;
+      }}
+    }}
+}}";
+
+        protected override IExecutionResult<TestResult> ExecuteAgainstTestsInput(
+            IExecutionContext<TestsInputModel> executionContext,
+            IExecutionResult<TestResult> result)
+        {
+            try
+            {
+                this.ExtractSubmissionFiles(executionContext);
+            }
+            catch (ArgumentException exception)
+            {
+                result.IsCompiledSuccessfully = false;
+                result.CompilerComment = exception.Message;
+
+                return result;
+            }
+
+            this.SaveTestsToFiles(executionContext.Input.Tests);
+            this.SaveNginxFile();
+
+            var codeSavePath = this.SavePythonCodeTemplateToTempFile();
+            var executor = this.CreateExecutor();
+            var checker = executionContext.Input.GetChecker();
+            return this.RunTests(codeSavePath, executor, checker, executionContext, result);
+        }
+
+        protected override IExecutionResult<TestResult> RunTests(
+            string codeSavePath,
+            IExecutor executor,
+            IChecker checker,
+            IExecutionContext<TestsInputModel> executionContext,
+            IExecutionResult<TestResult> result)
+        {
+            foreach (var test in executionContext.Input.Tests)
+            {
+                var processExecutionResult = this.Execute(executionContext, executor, codeSavePath, test.Input);
+                var testResults = this.ExtractTestResultsFromReceivedOutput(processExecutionResult, test.Id);
+                result.Results.AddRange(testResults);
+            }
+
+            return result;
+        }
```

Comment:
This can be refactored as:

```
 protected override IExecutionResult<TestResult> RunTests(
            string codeSavePath,
            IExecutor executor,
            IChecker checker,
            IExecutionContext<TestsInputModel> executionContext,
            IExecutionResult<TestResult> result)
  => executionContext.Input.Tests.Select(input => this.RunTest(executionContext, executor, codeSavePath, input);
  
  ```